packages/snippet-bot/src/violations.ts

```diff
@@ -198,10 +198,9 @@ const isAutoGenerated = (regionTag: string): boolean => {
 export const checkProductPrefixViolations = async (
   changes: ChangesInPullRequest,
   config: Configuration,
-  logger: GCFLogger = defaultLogger
+  apiLabels: ApiLabels
 ): Promise<Array<Violation>> => {
   const ret: Violation[] = [];
-  const apiLabels = await getApiLabels(dataBucket, logger);
```

Comment:
Avoid an implicit API call here -- explicitly provide the list of allowed products. The caller is expected to source the list.